XamarinCommunityToolkit/Helpers/BindablePropertyBuilder.shared.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+	public sealed class BindablePropertyBuilder
+	{
+		static readonly Lazy<BindablePropertyBuilder> lazyBindablePropertyBuilder = new Lazy<BindablePropertyBuilder>(() => new BindablePropertyBuilder());
+
+		readonly BindablePropertyParameters bpParameters;
+
+		BindablePropertyBuilder() => bpParameters = new BindablePropertyParameters();
+
+		/// <summary>
+		/// Resets the <see cref="BindablePropertyParameters"/> to its default values and
+		/// returns the <see cref="BindablePropertyBuilder"/> singleton
+		/// </summary>
+		/// <returns>The <see cref="BindablePropertyBuilder"/> singleton</returns>
+		public static BindablePropertyBuilder CreateBuilder()
+		{
+			var bpbSingleton = lazyBindablePropertyBuilder.Value;
+			bpbSingleton.bpParameters.ResetParameters();
```

Comment:
So we share a single builder among all bindable properties?
Wow, not sure it's a good idea due to threads race condition

Reply:
I don't think this is thread safe, if an app is loading 100s of Bindable Properties at once that exist in different controls I could see multiple threads being generated. Another scenario would be an app aggressively loading all pages of a tabbed dashboard at once.

Adding a simple lock will not solve this concurrency problem

## Additional Questions
- Whey can't we just instantiate a new `BindablePropertyParameters` each time a `CreateBuilder` is invoked? When I `CreateBuilder` I expect to have a fresh clean object.
- Why do we need the `BindablePropertyParameters`? I would prefer if we flattened the structure and just had instance properties in the builder class itself. If there is a good reason for this I can be convinced otherwise

Reply:
I think we should go back to the main issue #365 to settle this debate on static vs instance and performance.
Till we agree on the best solution, @aguevara716 won't have to reimplement the builder after each comment we make :)

Reply:
Thanks, @roubachof - I've actually been tempted to close this PR and issue and creating a new issue just so we can start the discussion from scratch.

That being said, I am actually learning a lot from everyone's comments! This is my first time contributing to an open source project, and the first time contributing to something that would be used by many people. I'm used to getting things to work for my use case and moving on, but this is definitely helping me to think more about how other developers would interact with my code.